mllib/src/main/scala/org/apache/spark/ml/util/SchemaUtils.scala

```diff
@@ -40,6 +40,8 @@ private[spark] object SchemaUtils {
       msg: String = ""): Unit = {
     val actualDataType = schema(colName).dataType
     val message = if (msg != null && msg.trim.length > 0) " " + msg else ""
+    require(actualDataType.getClass.getPackage.equals(dataType.getClass.getPackage),
+      s"Column $colName  of type ${dataType.typeName} must belong to ml package")
```

Comment:
I think you just want to improve the existing message; add the types that were checked. It doesn't need to be separate and that specific.

Reply:
Thanks for your review and advise, I will try to improve it.